exist-versioning-release.md

```diff
@@ -208,101 +313,169 @@ You will require a system with:
     What is the new development version for "eXist-db"? (org.exist-db:exist) 5.4.0-SNAPSHOT: :
```

Comment:
@windauer mentioned we should make it more explicit in the docs that the next version should always be a feature version. Do we want to expand the text here?

Maven by default will prompt you with the next patch version, i.e.:
```
What is the new development version for "eXist-db"? (org.exist-db:exist) 5.3.1-SNAPSHOT: :
```
So the docs should be updated here so the Maven prompt is correct. Even if you should reject its suggestion and then input the next feature version SNAPSHOT